Fix second+ image upload handling on CreateObs form

nimmolo opened the pull request:
This PR requires manual testing. To test, upload more than one image to a new observation.

  1. EXIF Extraction Fixes:
  - ✅ Server-side EXIF extraction for saved "good" images on edit forms
  - ✅ JavaScript optimization (skip client-side extraction for "good" images)
  - ✅ GPS visibility fix for owners viewing their own edit forms
  - ✅ Test fixture fallback for EXIF extraction in test environment
  - ✅ Type-safe GPS coordinates (Float instead of String)
  - ✅ Fallback to database values for images without GPS EXIF

  2. Test Improvements:
  - ✅ Combined two geotagged image tests into one to ensure predictable order
  - ✅ Added comprehensive EXIF extraction test for edit forms
  - ✅ Updated test helpers to handle hidden carousel indicators
  - ✅ Fixed flash message assertions (outdated expectations)
  - ✅ Updated Google Maps data constant (UNIVERSITY_PARK)

nimmolo committed:
Fix second+ image upload handling on CreateObs form

JoeCohen reviewed (commented):
I manually tested it by creating an Obs with 2 images, then editing an Obs by adding 2 images. It worked fine.
I don't understand phlex yet, so I don't understand the code. Sorry. I need to catch up on that soon.
The tests are failing in CI.

nimmolo committed:
Fix form camera info, form-exif stimulus and component test, system test

nimmolo committed:
Update form_camera_info.rb

nimmolo committed:
Fix exif extraction bugs

  1. itemTargetConnected now only skips if already initialized, not based on image status
  2. getExifData checks if image is already loaded before setting onload handler
  3. Extracts EXIF data immediately for cached/saved images

nimmolo committed:
Fix system test, more nilable props

nimmolo committed:
rubocop

mo-nathan commented on app/components/form_camera_info.rb:
```diff
@@ -23,13 +23,22 @@ class Components::FormCameraInfo < Components::Base
+  prop :img_id, _Nilable(String), &:to_s
+  # GPS coordinates can be passed as Float, Integer, or String
+  # Convert to string for display, handling blank values
+  prop :lat, _Nilable(String) do |v|
```
Not that are really care, but doesn't this completely skip around the type safety "feature"?  Seems like it would now work with any value that returns a string when `.to_s` is called on it.  If we're really trying to do type safety, then it seems like we should be fix the input side not the output side.  Put another way, why would `lat` ever need to be a String.  I assume it is under some circumstances currently, but rather than converting Floats to Strings shouldn't we converting all the Strings to a Float (or nil)?

nimmolo replied:
re: type safety  — These are all coming not from user input, but from the images via JS, and because of the laxity of the EXIF standard they could be strings, floats or integers. So i think it's fine to coerce them here.

re: how to type them. Claude initially was converting them to floats, i wonder why it changed it. Let me ask Claude ;)

nimmolo committed:
Really fix exif extraction and add notes and tests

nimmolo committed:
Fix camera info

  Bug Fixed: Camera Info was not displaying EXIF data (GPS, date, etc.) for saved "good" images when editing an observation.

  Root Causes Found:

  1. JavaScript was overwriting server data on this branch: The form-exif controller was trying to extract EXIF from resized browser-served images (which have EXIF stripped), overwriting the correct server-extracted data with empty values
  2. GPS hiding bug: read_exif_data was hiding GPS even from the owner viewing their own edit form
  3. Test environment limitation: Tests couldn't extract EXIF because process_image doesn't run in test mode

nimmolo committed:
simplify

nimmolo committed:
Update observation_form_system_test.rb

nimmolo committed:
Fix more system test assertions

nimmolo committed:
Update observation_form_system_test.rb

More system test fixes

nimmolo committed:
rubocop

mo-nathan commented on test/components/form_camera_info_test.rb:
```diff
@@ -26,10 +26,10 @@ def test_renders_gps_info_with_all_values
-    assert_includes(html, "45.5231</span>, ")
-    assert_includes(html, "-122.6765</span>, ")
-    assert_includes(html, "100</span>")
+    # Wrappers should not have d-none class when values are present
```
I don't understand this change.  What is the "d-none class"?  In what sense are the following assertions checking that some class is NOT present?  Why are we now just testing to see if a class is mentioned vs. numeric values?
